src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -237,6 +245,18 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 
 		$options = wp_parse_args( $options, array( 'with_supports' => true ) );
 
+		$can_use_cached = ! wp_is_development_mode( 'theme' );
+		if ( $can_use_cached ) {
+			$cache_key = 'get_theme_data_cache_' . md5( wp_json_encode( func_get_args() ) );
```

Comment:
I don't think using `func_get_args` here is necessary, because the theme data get's cached prior to when the `with_supports` option is calculated.

Reply:
I have used `$options` instead in the new update as we don't quite need other parameters. At the moment set_transient is getting called on every run which is the reason why we don't see any benefit. Comparison of two static variables is not working quite well also other interesting find (off topic) I am able to serialize those variables but not convert into JSON

Reply:
@joemcgill You were correct we don't need $options too.